truex-ios-mobile-reference-app/UnlockSoloViewController.m

```diff
@@ -0,0 +1,313 @@
+    NSLog(@"truex: onAdFreePod");
+    [self.unlocked setOn:YES];
+}
+
+// [5] - Other delegate method
+- (void)onPopupWebsite:(NSString *)url {
+    // true[X] - User wants to open an external link in the true[X] ad
+
+    NSLog(@"truex: onPopupWebsite: %@", url);
+    // Open URL with the SFSafariViewController
+    // SFSafariViewController *svc = [[SFSafariViewController alloc] initWithURL:[NSURL URLWithString: url]];
+    // svc.delegate = self;
+    // svc.modalPresentationStyle = UIModalPresentationOverCurrentContext;
+    // [self presentViewController:svc animated:YES completion:nil];
+    // [self.activeAdRenderer pause];
+    
+    // Or, open the URL directly in Safari
+    // [[UIApplication sharedApplication] openURL:[NSURL URLWithString: url] options:@{} completionHandler:nil];
+    
+    // Or, open with the existing in-app webview
+     UIStoryboard* storyBoard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
+     WebViewViewController* newViewController = [storyBoard instantiateViewControllerWithIdentifier:@"webviewVC"];
+     newViewController.url = [NSURL URLWithString:url];
+     newViewController.modalPresentationStyle = UIModalPresentationOverCurrentContext;
+     __weak typeof(self) weakSelf = self;
+     newViewController.onDismiss = ^(void) {
+         // true[X] - You will need to pause and remume the true[X] Ad Renderer
+         [weakSelf.activeAdRenderer resume];
+     };
+     [self.activeAdRenderer pause];
+     [self presentViewController:newViewController animated:YES completion:nil];
+}
+
+// When using SFSafariViewController for onPopupWebsite
+//- (void)safariViewControllerDidFinish:(SFSafariViewController *)controller {
+//    // true[X] - You will need remume the true[X] Ad Renderer after safariViewController
+//    if (self.activeAdRenderer) {
+//        [self.activeAdRenderer resume];
+//    }
+//}
+
+
+// MARK: - Helper Functions / Fake Ad Framework
+- (void)fetchAd:(NSString *)url {
+    if (self.vastDictionary != nil) {
+        return;
+    }
+    url = [url stringByReplacingOccurrencesOfString:@"[stream_id]" withString:[[NSUUID UUID] UUIDString]];
+    // replacing user_id with random UUID here for testing, please user the real user ID from the system.
+    url = [url stringByReplacingOccurrencesOfString:@"[user_id]" withString:[[NSUUID UUID] UUIDString]];
+    
+    // Fetch the xml from server
+    NSXMLParser *xmlparser = [[NSXMLParser alloc] initWithContentsOfURL:[[NSURL alloc] initWithString:url]];
+    
+    // Or use the hardcoded copy
+    // NSData* vastSoloData = [NSData dataWithContentsOfFile:[[NSBundle mainBundle] pathForResource:@"vast_solo" ofType:@"xml"]];
+    // NSXMLParser *xmlparser = [[NSXMLParser alloc] initWithData:vastSoloData];
+    
+    [xmlparser setDelegate:self];
+    BOOL success = [xmlparser parse];
+    if (success) {
+//        // Print vastDictionary as JSON string to help with debug.
```

Comment:
Why is this commented out?

Reply:
To help anyone trying to debug/understand this code. 
It is too noisy to always print this out and it is not part of the reference app. 
Commenting that piece out to me represents the above idea.